### PR TITLE
Added a password component.

### PR DESCRIPTION
**Pull request by Imnaseli**
Feature addition: Password component

I added a password component.
I found out chakra dont have a separate password component, the type is just changed between "text" and "password".
so i simply imported the input component in a new class and changed the "type_" to password.
so that instead of pc.input(type_="password") users can just use pc.password().

### CODE REVIEW

**Review by picklelo — APPROVED** [review]
Awesome

File: pynecone/components/forms/password.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from typing import Set
+from .input import Input
+from pynecone.components.component import EVENT_ARG
+from pynecone.var import Var
+
+
+class Password(Input):
+    """The Password component is a component that is used to get password input from users."""
+
+    type_: Var[str] = "password"
```
Comment (picklelo) [inline pynecone/components/forms/password.py:12]: I think you need a `#type: ignore` at the end of this line to make Pyright happy

**Review by picklelo — CHANGES_REQUESTED** [review]

File: pynecone/components/forms/password.py (hunk 1 above)
Comment (Imnaseli) [reply pynecone/components/forms/password.py:12]: Okay will do that.

Comment (Imnaseli) [reply pynecone/components/forms/password.py]: I fixed it and pyright is now happy :D

**Review by picklelo — CHANGES_REQUESTED** [review]
I think there's still lint errors: make sure all the checks in the [Contributing guide](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) pass

**Comment by Imnaseli** [comment]
im sorry i keep getting an error concerning the 'TestState' everytime i test it with pytest.
`
cannot collect test class 'TestState' because it has a __init__ constructor (from: tests/test_state.py)
 `
i believe it is the same error that the pull request of  #437 is trying to solve.
should i just close the pull request.

**Comment by picklelo** [comment]
> im sorry i keep getting an error concerning the 'TestState' everytime i test it with pytest.
> `cannot collect test class 'TestState' because it has a __init__ constructor (from: tests/test_state.py)`
> i believe it is the same error that the pull request of #437 is trying to solve.
> should i just close the pull request.

The tests look like they're all passing in the Github Actions?

I think you just have to run
```
poetry run black pynecone tests
```

to lint the files, and it should be good to go!

**Pull request closed without merging**